src/components/test_editor.rs

```diff
@@ -1979,6 +1979,34 @@ fn entering_insert_mode_from_visual_mode() -> anyhow::Result<()> {
     })
 }
 
+#[test]
+fn modifying_editor_causes_dirty_state() -> anyhow::Result<()> {
+    execute_test(|s| {
+        Box::new([
+            App(OpenFile(s.main_rs())),
+            Expect(Not(Box::new(EditorIsDirty()))),
+            Editor(EnterInsertMode(Direction::Start)),
+            App(HandleKeyEvents(keys!("a a esc").to_vec())),
```

Comment:
Better use `DispatchEditor`s rather than `keys!` so that these test cases are not affected by the keymap changes.

Reply:
Oops, I didn't notice that's in Insert mode.